doc/design/selected_rows.md

```diff
@@ -0,0 +1,56 @@
+# Design Doc: Selected Rows
+
+`SelectedRows` is a kind of sparse tensor data type, which is designed to support `embedding` operators. The gradient of embedding table is a sparse tensor. Only a few rows are non-zero values in that tensor. It is straightforward to represent the sparse tensor by the following sparse tensor data structure:
+
+```cpp
+class SelectedRows {
+ private:
+  vector<int> rows_;
+  Tensor value_;
```

Comment:
Giving an example of SelectedRows is helpful to understand.
We have a 2-D np array data,
```
[[ 1, 2, 3],
 [ 0, 0, 0],
 [ 0, 0, 0],
 [ 4, 5, 6],
 [ 0, 0, 0]]
```
So, how will it be represented in SelectedRows

Reply:
The example has been given in [Line 16](https://github.com/PaddlePaddle/Paddle/pull/4652/files#diff-879e40df9669ca7b37eb6ae84cc5f359R16).